qutip/core/coefficient.py

```diff
@@ -208,6 +208,10 @@ class CompilationOptions:
         "extra_import": ""
     }
 
+# Version number of the Coefficient
+# Str without '.'
+COEFF_VERSION = "1_0"
```

Comment:
@jakelishman
I added coefficient version here mixed with `brmesolve` (yes my branches are a mess...).
I did not add it as salting the hast but added it directly, then we can add a purge old coeff functions.
I did not use the qutip version but a coeff version so that we don't hve to recompile when this part of the code is untouched.

Reply:
Nice - definitely a nice touch.

I'd be a little worried that at some point in the future we'll forget to update this variable when the code changes, and get something wrong in a release.  Having it tied to the QuTiP version means we'll never fail on that front, but at the cost that recompilations will be triggered whenever the user updates QuTiP - I wouldn't see that as a huge negative, because QuTiP releases aren't very common.

For purging purposes, it's maybe easier to create subdirctories inside the coefficient one that include the full (coeff or qutip) version and avoid storing it in the name?  It avoids problems with having to glob to find invalid files.  You can use a regular version string (even `+` is allowed in directory names on everything) and use regular version-parsing systems, so it'd be easier for a purger to (say) delete only files from older versions of QuTiP but not newer ones.